src/service/resources-impl.js

```diff
@@ -387,15 +388,40 @@ export class Resources {
     this.resources_.push(resource);
 
     if (this.isRuntimeOn_) {
-      // Try to immediately build element, it may already be ready.
-      resource.build(this.forceBuild_);
-      this.schedulePass();
+      if (this.documentReady_) {
+        // Try to immediately build element, it may already be ready.
+        resource.build();
+        this.schedulePass();
+      } else {
+        this.pendingBuildResources_.push(element);
+        this.buildReadyResources_();
+      }
     }
 
     dev.fine(TAG_, 'element added:', resource.debugid);
   }
 
   /**
+   * Builds resources that are ready to be built.
+   * @private
+   */
+  buildReadyResources_() {
+    const builtResourcesIndexes = [];
+    this.pendingBuildResources_.forEach((resource, index) => {
```

Comment:
Avoids a function allocation, avoids edge cases with holes that `#forEach` has to deal with. In general, SPEED. 🐎 🏇